### DISCUSSION TITLE
[A-L] (10 settembre 2024) Esercizio 3

### DISCUSSION THREAD

**Discussion by FeddyLix17** [sapienzastudentsnetwork/progettazione-di-algoritmi#116] [Esami A-L]
Dato un insieme di intervalli chiusi $\Large [a_i,\ b_i],\ 1 \leq i \leq k$, trovare un sottoinsieme massimale di intervalli che si intersettono ciascuno.

Per esempio, dato $\Large \\{[1,2], [3,5], [4,5], [5,7]\\}$ l'algoritmo restituisce $\Large \\{[3,5], [4,5], [5,7]\\}$.

- **Suggerimento**: NON pensare su un algoritmo greedy.

**Comment by niccolo-fato** [comment]
![Screenshot 2025-05-24 alle 21 13 57](https://github.com/user-attachments/assets/37c5bdf6-a528-400d-8998-7ac9c99f76cb)
![Screenshot 2025-05-24 alle 21 14 12](https://github.com/user-attachments/assets/cafb6fa3-7601-4bea-baf0-ac7278cc207d)
